### ISSUE TITLE
How does keras aggregate pixelwise losses into its the single-number loss value?

### ISSUE THREAD

**Issue by felix-tracxpoint** [keras-team/keras#11627]
When training a semantic segmentation network with the loss defined [here](https://gist.github.com/wassname/f1452b748efcbeb4cb9b1d059dce6f96) I get negative numbers in the keras printout:
```

Epoch 1/20
3900/3900 [==============================] - 4421s 1s/step - loss: -8.9678 - acc: 0.0660
Epoch 2/20
3900/3900 [==============================] - 4331s 1s/step - loss: -34.6251 - acc: 0.0643
Epoch 3/20
3900/3900 [==============================] - 4315s 1s/step - loss: -81.8369 - acc: 0.0958
Epoch 4/20
3900/3900 [==============================] - 4349s 1s/step - loss: -169.9554 - acc: 0.0360
Epoch 5/20
3900/3900 [==============================] - 4277s 1s/step - loss: -328.2449 - acc: 0.0278
Epoch 6/20
3900/3900 [==============================] - 4243s 1s/step - loss: -636.5703 - acc: 0.0258
```

This is rather weird because the loss function generates **non-negative** values per pixel. I don't quite understand how keras aggregates them into its single number but however it does it, the result should not be negative, should it?

I suppose I am missing something huge here.

Many thanks in advance for your advice!
